### PR TITLE
Show referral impressions and CTRs

### PR DESCRIPTION
**Pull request by kelvin-chappell**
This adds two extra columns to the referrals table:
image

### CODE REVIEW

File: app/model/CampaignReferral.scala [hunk 1]
```diff
@@ -20,11 +20,29 @@ object Component {
 // A referral to any item in a campaign from an on-platform source in a particular period
 case class CampaignReferral(
   component: Component,
-  numClicks: Int,
+  clickCount: Int,
+  impressionCount: Int,
+  // Annoyingly react won't recognise CTR if it's a derived property so it has to go in as part of the constructor
```
Comment (LATaylor-guardian) [inline app/model/CampaignReferral.scala]: I don't understand what you mean here, can you explain please?

**Comment by LATaylor-guardian** [comment]
👍

File: app/model/CampaignReferral.scala (hunk 1 above)
Comment (kelvin-chappell) [reply app/model/CampaignReferral.scala]: Thanks for questioning it.  I realise that it makes no sense at all, and I've fixed the underlying problem now.

Comment (LATaylor-guardian) [reply app/model/CampaignReferral.scala]: Great :)

**Pull request merged**